istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,564 @@
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enable webhook configurations",
+		Example: `
+# Enable the webhook configuration of Galley with the given webhook configuration
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --config-path validatingwebhookconfiguration.yaml
+
+# Enable the webhook configuration of Galley with the given webhook configuration and CA certificate
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --config-path validatingwebhookconfiguration.yaml --ca-bundle-file ./k8s-ca-cert.pem
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// --namespace is used as the namespace of the validation secret
+			opts.validationSecretNameSpace = namespace
+			if len(opts.webhookConfigPath) == 0 {
+				return fmt.Errorf("must specify a valid --config-path")
+			}
+			if len(opts.validationSecretName) == 0 {
+				return fmt.Errorf("must specify a valid --validation-secret")
+			}
+			if !opts.enableValidationWebhook {
+				return fmt.Errorf("not enabling validation webhook")
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating Kubernetes client interface: %v", err)
+			}
+			// Read Kubernetes CA certificate that will be used as the CA bundle of the webhook configuration
+			caCert, err := readCACert(client, opts.caCertPath, opts.validationSecretName, opts.validationSecretNameSpace,
+				opts.maxTimeForReadingWebhookCert)
+			if err != nil {
+				return fmt.Errorf("err when reading CA certificate: %v", err)
+			}
+			fmt.Printf("Webhook CA certificate:\n%v\n", string(caCert))
+			// Apply the webhook configuration when the Galley webhook server is running.
+			err = waitForServerRunning(client, namespace, opts.validatingWebhookServiceName,
+				opts.maxTimeForCheckingWebhookServer)
+			if err != nil {
+				return fmt.Errorf("err when checking webhook server: %v", err)
+			}
+			webhookConfig, err := buildValidatingWebhookConfig(
+				caCert,
+				opts.webhookConfigPath,
+				opts.validatingWebhookConfigName,
+			)
+			if err != nil {
+				return fmt.Errorf("err when build validatingwebhookconfiguration: %v", err)
+			}
+			_, err = createValidatingWebhookConfig(client, webhookConfig)
+			if err != nil {
+				return fmt.Errorf("error when creating validatingwebhookconfiguration: %v", err)
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.enableValidationWebhook, "validation", true, "Enable "+
+		"validating webhook (default true).")
+	// Specifies the local file path to webhook CA bundle.
+	// If empty, will read CA bundle from default Kubernetes service account.
+	flags.StringVar(&opts.caCertPath, "ca-bundle-file", "",
+		"Certificate signing CA bundle .pem file. "+
+			"If the certificate is signed by Kubernetes CA, this parameter needs to "+
```

Comment:
"PEM encoded CA bundle which will be used to validate the webhook's server certificates. If this is empty, the kube-apisever's root CA is used if it can be confirmed to have signed the webhook's certificates. This condition is sometimes true but is not guaranteed (see https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping)"

Reply:
Done: changed to the text in the comment.